teps/0056-pipelines-in-pipelines.md

```diff
@@ -485,7 +495,242 @@ These are alternatives to the solution for status discussed above:
 
 ## Design
 
-TODO
+In this section, we flesh out the details of `Pipelines` in `Pipelines` in relation to:
+- [Parameters](#parameters)
+- [Results](#results)
+  - [Consuming Results](#consuming-results)
+  - [Producing Results](#producing-results)
+- [Execution Status](#execution-status)
+- [Workspaces](#workspaces)
+- [When Expressions](#when-expressions)
+- [Retries](#retries)
+- [Timeouts](#timeouts)
+- [Matrix](#matrix)
+
+### Parameters
+
+`Pipelines` in `Pipelines` will consume `Parameters` in the same way as `Tasks` in `Pipelines`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  params:
+    - name: repo
+      value: $(params.repo)
+  tasks:
+    - name: git-clone
+      params:
+        - name: repo
+          value: $(params.repo)      
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      params:
+        - name: repo
+          value: $(params.repo)
+      pipelineRef:
+        name: security-scans
+    - name: notification
+      taskRef:
+        name: notification
+```
+
+### Results
+
+#### Consuming Results 
+
+`Pipelines` in `Pipelines` will consume `Results`, in the same way as `Tasks` in `Pipelines`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  params:
+    - name: repo
+      value: $(params.repo)
+  tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      params:
+        - name: commit
+          value: $(tasks.git-clone.results.commit) # --> result consumed in pipelines in pipelines
+      pipelineRef:
+        name: security-scans
+    - name: notification
+      taskRef:
+        name: notification
+```
+
+#### Producing Results
+
+`Pipelines` in `Pipelines` will produce `Results`, in the same way as `Tasks` in `Pipelines`.
+
+`Results` produced by `TaskRuns` in a child `PipelineRun` that need to be passed to subsequent `PipelineTasks` will need
+to be propagated to the `Results` of the child `PipelineRun`.
```

Comment:
I hope eventually we'll do for results something similar to parameters propagation :D

Reply:
💯 results propagation was listed in future work in TEP-0107: https://github.com/tektoncd/community/blob/main/teps/0107-propagating-parameters.md#passing-results